Refactor RenderReplaced::computeConstrainedLogicalWidth.

sammygill opened the pull request:
#### 68309d190f8639e836883cfc26abf23ceb5b85fc
<pre>
Refactor RenderReplaced::computeConstrainedLogicalWidth.
<a href="https://bugs.webkit.org/show_bug.cgi?id=282417">https://bugs.webkit.org/show_bug.cgi?id=282417</a>
<a href="https://rdar.apple.com/problem/139034753">rdar://problem/139034753</a>

Reviewed by Vitor Roriz.

Small refactoring that should have no functional change in preparation for a bug
fix.

This method does a couple of more things than the name implies. It has
some special logic if it is called when we are computing the preferred logical
widths and also calls computeReplacedLogicalWidthRespectingMinMaxWidth to try and
do some clamping.

Let&apos;s refactor this method so that it only does what it states in the name by moving
the responsibility of performing this extra logic to the callers. There are 2 call
sites where this change is needed.

* Source/WebCore/rendering/RenderReplaced.cpp:
(WebCore::RenderReplaced::paint):
* Source/WebCore/rendering/RenderReplaced.h:

Canonical link: <a href="https://commits.webkit.org/286155@main">https://commits.webkit.org/286155@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fcf58835a8ccb66e5c826e348056b2cd6164412d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/74947 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/54377 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/27764 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/79404 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/26186 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/77064 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/63513 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/2162 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/58860 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/17118 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/78014 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/49000 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/64389 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/39250 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/46301 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/21888 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/24518 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/67444 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/22232 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/80863 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/2265 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/1367 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/67100 "Found 3 new test failures: imported/w3c/web-platform-tests/css/css-view-transitions/inline-child-with-filter.html imported/w3c/web-platform-tests/css/css-view-transitions/new-content-object-view-box-clip-path.html imported/w3c/web-platform-tests/css/css-view-transitions/new-content-object-view-box-overflow.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/2414 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/64408 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/66400 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16511 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/10337 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/8500 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/2230 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/5018 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/2258 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/3179 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/2265 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

vitorroriz commented on Source/WebCore/rendering/RenderReplaced.cpp:
```diff
@@ -634,8 +640,12 @@ LayoutUnit RenderReplaced::computeReplacedLogicalWidth(ShouldComputePreferred sh
+                if (shouldComputePreferred == ComputePreferred)
+                    return computeReplacedLogicalWidthRespectingMinMaxWidth(0_lu, ComputePreferred);
+                auto constrainedLogicalWidth = computeConstrainedLogicalWidth();
+                return computeReplacedLogicalWidthRespectingMinMaxWidth(constrainedLogicalWidth, ComputeActual);
```
I understand that you want to make `computeConstrainedLogicalWidth` do what it is saying it does, but don't you want to abstract this repeated logic in another lambda/function with a significant name?

sammygill replied:
The constrained logical width computation should be completely separate from any preferred logical widths logic so factoring that out into the same function would be just as wrong. 

Aside from that we could refactor the constrained logical width computation and then the call to `computeReplacedLogicalWidthRespectingMinMaxWidth` into something but we already have similar functions like that on render tree and they are honestly confusing because it's really hard to tell when you should call which one :/

sammygill committed:
Refactor RenderReplaced::computeConstrainedLogicalWidth.
https://bugs.webkit.org/show_bug.cgi?id=282417
rdar://problem/139034753

Reviewed by Vitor Roriz.

Small refactoring that should have no functional change in preparation for a bug
fix.

This method does a couple of more things than the name implies. It has
some special logic if it is called when we are computing the preferred logical
widths and also calls computeReplacedLogicalWidthRespectingMinMaxWidth to try and
do some clamping.

Let's refactor this method so that it only does what it states in the name by moving
the responsibility of performing this extra logic to the callers. There are 2 call
sites where this change is needed.

* Source/WebCore/rendering/RenderReplaced.cpp:
(WebCore::RenderReplaced::paint):
* Source/WebCore/rendering/RenderReplaced.h:

Canonical link: https://commits.webkit.org/286155@main
